packages/orbit-components/src/List/ListItem/index.tsx

```diff
@@ -1,143 +1,28 @@
-import { SIZES, TYPES } from "../consts";
-import { StyledText } from "../../Text";
-import ListContext from "../ListContext";
 import type { Props } from "./types";
 
-export const getLineHeightToken = ({
-  theme,
-  size,
-}: {
-  theme: Theme;
-  size?: Common.Size | null;
-}): string | null => {
-  const lineHeightTokens = {
-    [SIZES.SMALL]: theme.orbit.lineHeightTextSmall,
-    [SIZES.NORMAL]: theme.orbit.lineHeightTextNormal,
-    [SIZES.LARGE]: theme.orbit.lineHeightTextLarge,
-  };
-  if (!size) return null;
-  return lineHeightTokens[size];
-};
-
-const getSizeTokenLabel = ({ theme, size }) => {
-  const sizeTokens = {
-    [SIZES.SMALL]: theme.orbit.fontSizeTextSmall,
-    [SIZES.NORMAL]: theme.orbit.fontSizeTextSmall,
-    [SIZES.LARGE]: theme.orbit.fontSizeTextNormal,
-  };
-  return sizeTokens[size];
-};
-
-const getIconSizeFromType = ({
-  theme,
-  type,
-}: {
-  theme: Theme;
-  type?: Type | null;
-}): string | null => {
-  const tokens = {
-    [TYPES.PRIMARY]: css`
-      height: ${theme.orbit.heightIconSmall};
-      width: ${theme.orbit.widthIconSmall};
-    `,
-    [TYPES.SECONDARY]: css`
-      height: ${theme.orbit.heightIconSmall};
-      width: ${theme.orbit.widthIconSmall};
-    `,
-  };
-
-  if (!type) return null;
-  return tokens[type];
-};
-
-export const Item = styled(({ ...props }) => <li {...props} />)`
-  ${({ theme }) => css`
-    font-family: ${theme.orbit.fontFamily};
-    display: flex;
-    flex-direction: row;
-    margin-bottom: ${theme.orbit.spaceXXSmall};
-
-    &:last-child,
-    &:last-of-type {
-      margin: 0;
-    }
-
-    ${StyledText} {
-      line-height: inherit;
-      font-size: inherit;
-    }
-  `}
-`;
-
-Item.defaultProps = {
-  theme: defaultTheme,
-};
-
-export const IconContainer = styled.div<{ size?: Common.Size | null; type?: Type | null }>`
-  ${({ theme, size }) => css`
-    display: flex;
-    align-items: center;
-    margin: ${rtlSpacing(`0 ${theme.orbit.spaceXSmall} 0 0`)};
-    flex: 0 0 auto;
-    height: ${getLineHeightToken({ theme, size })};
-
-    ${StyledCarrierLogo} {
-      ${getIconSizeFromType};
-      img {
-        ${getIconSizeFromType};
-      }
-    }
-    svg {
-      ${getIconSizeFromType};
-    }
-  `}
-`;
-
-IconContainer.defaultProps = {
-  theme: defaultTheme,
-};
-
-const StyledLabel = styled.div`
-  ${({ theme }) => css`
-    font-family: ${theme.orbit.fontFamily};
-    font-weight: ${theme.orbit.fontWeightNormal};
-    color: ${theme.orbit.colorTextSecondary};
-    font-size: ${getSizeTokenLabel};
-  `}
-`;
-
-const StyledSpan = styled.span`
-  width: 100%;
-`;
-
-StyledLabel.defaultProps = {
-  theme: defaultTheme,
-};
-
 const ListItem = ({ label, children, icon = <CircleSmall />, dataTest }: Props) => {
-  const { size, type } = React.useContext(ListContext);
   return (
-    <Item data-test={dataTest} type={type}>
+    <li
+      data-test={dataTest}
+      className="orbit-list-item font-base mb-xxs flex last:m-0 last-of-type:m-0"
```

Comment:
imo no, as long as it looks okay I wouldn't add redundant styles. `flex-row` is the default, so not necessary

why does it need `StyledText` / `.orbit-text`? colour is handled level up, font here (though should not be necessary imo)

Reply:
> imo no, as long as it looks okay I wouldn't add redundant styles

I agree. My concern is that I am not sure if it wasn't there in the first place because it was needed because of something else. But I am in favor of avoiding extra redundant styles.

Regarding Text, it was applying `inherit` to `line-height` and `font-size` properties so that it is consistent with the size passed to List, I assume